addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -0,0 +1,424 @@
+                                                  Response response,
+                                                  final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        if ( transfer == null )
+        {
+            return;
+        }
+
+        OutputStream out1 = null;
+        OutputStream out2 = null;
+
+        Transfer target1;
+        Transfer target2;
+        String content1 = "";
+        String content2 = "";
+
+        ConcreteResource resource = transfer.getResource();
+        try
+        {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode root = mapper.readTree( transfer.openInputStream( true ) );
+
+            String path1 = null;
+            String path2 = null;
+            JsonNode vnode = root.path( "versions" );
+            JsonNode anode = root.path( "_attachments" );
+            JsonNode idnode = root.path( "_id" );
+
+            if ( vnode.fields().hasNext() )
+            {
+                String version = vnode.fields().next().getKey();
+                if ( version == null )
+                {
+                    return;
+                }
+                path1 = Paths.get( idnode.asText(), version ).toString();
+                content1 = vnode.findValue( version ).toString();
+            }
+
+            if ( anode.fields().hasNext() )
+            {
+                String tarball = anode.fields().next().getKey();
+                if ( tarball == null )
+                {
+                    return;
+                }
+                path2 = Paths.get( idnode.asText(), "-", tarball ).toString();
+                content2 = anode.findPath( "data" ).asText();
+            }
+
+            if ( path1 == null || path2 == null )
+            {
+                return;
+            }
+            target1 = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), path1 ) );
+            logger.info( "STORE {}", target1.getResource() );
+
+            target2 = transfers.getCacheReference( new ConcreteResource( resource.getLocation(), path2 ) );
+            logger.info( "STORE {}", target2.getResource() );
+        }
+        catch ( final IOException e )
+        {
+            logger.error( String.format( "[NPM] Json node parse failed for resource: %s. Reason: %s", resource,
+                                         e.getMessage() ), e );
+            response = formatResponse( e, builderModifier );
+
+            return;
+        }
+        try
+        {
+            if ( target1 == null || target2 == null )
+            {
+                return;
+            }
+            out1 = target1.openOutputStream( TransferOperation.UPLOAD, true, eventMetadata );
+            out1.write( content1.getBytes() );
+
+            out2 = target2.openOutputStream( TransferOperation.UPLOAD, true, eventMetadata );
+            out2.write( Base64.decodeBase64( content2 ) );
+
+            setGeneratedTransfers( transfer, target1, target2 );
```

Comment:
Please return this from the method instead of using a static variable or ThreadLocal or anything like that. Even if you have to create an internal subclass data type to return everything needed from this method, it's better than using static maps...If anything happens to go wrong and we're using a static map, those values may never get cleared out.

Reply:
Not fixed in new code?